jax/_src/lax/control_flow.py

```diff
@@ -383,43 +383,77 @@ def _while_loop_batching_rule(args, dims, axis_name, main_type,
   size, = {x.shape[d] for x, d in zip(args, dims) if d is not batching.not_mapped}
   orig_batched = [d is not batching.not_mapped for d in dims]
   cconst_bat, bconst_bat, init_bat = split_list(orig_batched, [cond_nconsts, body_nconsts])
+  cconsts, bconsts, init = split_list(args, [cond_nconsts, body_nconsts])
+  cconst_dims, bconst_dims, init_dims = split_list(dims, [cond_nconsts, body_nconsts])
 
+  carry_bat = init_bat
   # Fixpoint computation of which carry are batched: either
   # batched from init, or the carry out is batched. Each iteration promotes
   # at least one carry to batched. We need at most len(carry) iterations,
   # but we need one last iteration to prepare the jaxpr based on the final
   # carry_bat.
```

Comment:
This comment needs an update (e.g. the details of last iteration no longer hold)

Reply:
Done.